plugins/bundle/plugin.go

```diff
@@ -310,10 +313,14 @@ func (p *Plugin) UnregisterBulkListener(name interface{}) {
 
 // Config returns the plugins current configuration
 func (p *Plugin) Config() *Config {
+	p.cfgMtx.RLock()
+	defer p.cfgMtx.RUnlock()
```

Comment:
We aren't making a copy of the config, so what benefit is the lock giving us here?

Reply:
I am thinking we still need to protect against reading while writes are in-progress.

Reply:
Just understood the point. Will correct.